### PR TITLE
EPIC MUSTACHES MOVIES LIST! - Johanna Blom

### PR DESCRIPTION
**Pull request by idsintehittapa**
**Link to my deployed project:**
https://epic-mustaches.netlify.app 

**My reflections on how this week's project turned out:**
I am very happy with this weeks project! 
- I managed to make my own list
- I also managed (with help) to make the API key not public
- I added a footer and some fun svgs. 

The project has/is
🔵  Blue Level (Minimum Requirements)**

- at least two pages - one showing a list of movies and one showing details
- following the design from our example (but it's ok to change things - just try to make it look nice)

🔴  Red Level (Intermediary Goals)**

- Showing a 'not found' page if you try to visit a movie detail page with an invalid movie ID
- Handles loading states on going from movie list to movie details.

### CODE REVIEW

File: code/src/components/Header.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export const Header = () => {
+  return (
+    <header>
+      <h1 className='page-title'>Epic mustaches movies list!</h1>
+      <Route path='/movies'>
+      <Link to={"/"}>
+      <span className='mustache-icon'><Mustache /><p className='go-back'>Go back to movies</p></span>
+        </Link>
+      </Route>
```
Comment (ylvan75) [inline code/src/components/Header.js:16]: I like that you have a header and that you using this method to decide that some content shall only be presented at `<Route path='/movies'> `

File: code/src/components/Header.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+.page-title{
+  color: #fff;
+  text-transform: uppercase;
+  margin: 20px;
+}
+
+@media screen and (min-width:768px){
+  .mustache-icon{
+    /* margin-top: 2.5vh; */
```
Comment (ylvan75) [inline code/src/components/Header.css:33]: This can removed.

**Review by ylvan75 — APPROVED** [review]
Hi Johanna :-) 

I am always impressed by your creativity as in this case to have the theme "Epic mustaches movies list!"

Cool that you made your own list and that you also fixed the API key to not be public.

It was perfect with the information (pull request template) in the PR and keep the README -file short and clean. I will think about that for my project's
further on.
 
Such a great 404 page - Love it ❤️

I agree with your comment on my PR that is was really hard to peer review this since our solutions are so similar.

I have made some comments and they are mainly of the type good practice.

My last thing I wanna say is that I looked at you site at these browsers/device, see list, and you site worked really well. 
-Chrome - Mac
-Firefox - Windows
-Edge - Windows
-iOS - iPone X (Safari)
-Android - Samsung S8 (Chrome)

Great job!

File: code/src/components/Footer.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
+import './Footer.css'
+
+export const Footer = () => {
+  return (
+    <footer>
+      <p className='footer-info'> This is a project made via the Technigo front-end bootcamp, Fall 2020. 
+      <br></br>My <a href='https://github.com/idsintehittapa'>GitHub</a>
```
Comment (EmOlsz) [inline code/src/components/Footer.js:9]: We shouldn't use `br` tags at all, needs to be fixed

File: code/src/components/Header.css [hunk 4]
```diff
@@ -0,0 +1,43 @@
+header{
+  display: flex;
+  align-items: center;
+}
+
+.mustache-icon{
+  display:flex;
+  flex-direction: row;
+  align-items:center;
+  /* margin-top: 5vh; */
```
Comment (EmOlsz) [inline code/src/components/Header.css:10]: Before final push, remove all commented code

File: code/src/components/Header.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import { ReactComponent as Mustache } from 'media/mustache.svg'
+
+import './Header.css'
+
+export const Header = () => {
+  return (
+    <header>
+      <h1 className='page-title'>Epic mustaches movies list!</h1>
+      <Route path='/movies'>
+      <Link to={"/"}>
```
Comment (EmOlsz) [inline code/src/components/Header.js:13]: `<Link to="/" />` is completely enough

File: code/src/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      <span className='mustache-icon'><Mustache /><p className='go-back'>Go back to movies</p></span>
```
Comment (EmOlsz) [inline code/src/components/Header.js:14]: It's tricky business to nest anything inside `span` elements - they are inline, so a lot of styles of children might be messed up. Consider using `span` only in the final level (so, without any nesting below) and replace it over here with `div`

File: code/src/pages/PageNotFound.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+
+import { Footer } from 'components/Footer'
```
[the remark this answers is not in the document]
Comment (idsintehittapa) [reply code/src/pages/PageNotFound.js:3]: They are in different folders - that's why I did not use ./. The path ./componentes/Footer does not exist 😄

File: code/src/pages/MovieDetails.css [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+  display:flex;
+  flex-direction: column;
+  align-items: center;
+  margin-top:50px;
+}
+
+
+@media screen and (min-width:768px) {
+  .movie-thumb-wrapper{
+    grid-template-columns: repeat(2, 1fr);
```
[the remark this answers is not in the document]
Comment (idsintehittapa) [reply code/src/pages/MovieDetails.css:71]: You know Ylva, this does not actually do anything :O - I tried to check now but it's redundant so I removed it. Thanks for finding it 😄

File: code/src/components/MovieThumb.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+import './MovieThumbs.css'
+
+export const MovieThumb = ({ id, poster_path, title, release_date }) => {
+  const year = new Date(release_date).getFullYear()
+  return (
+    <article className='movie-thumb'>
+      <Link to={`/movies/${id}`}>
+        <img
+          className='movie-poster'
+          src={`https://image.tmdb.org/t/p/w342/${poster_path}`} // check this
```
[the remark this answers is not in the document]
Comment (idsintehittapa) [reply code/src/components/MovieThumb.js:13]: Yeah, sorry!

File: code/src/components/Footer.js (hunk 3 above)
Comment (idsintehittapa) [reply code/src/components/Footer.js:9]: Fixed

**Comment by idsintehittapa** [comment]
I've updated the project with the requested changes

**Pull request closed without merging**